client/my-sites/comments/comment-list/index.jsx

```diff
@@ -206,6 +207,29 @@ export class CommentList extends Component {
 		this.props.undoBulkStatus( selectedComments );
 	}
 
+	getPlaceholderOrEmptyContent() {
+		const { isLoading, comments, siteId } = this.props;
+		const [ emptyMessageTitle, emptyMessageLine ] = this.getEmptyMessage();
+		const zeroComments = size( comments ) <= 0;
+		if ( ( ! siteId || isLoading ) && zeroComments ) {
+			return (
+				<CommentDetailPlaceholder key="comment-detail-placeholder" />
+			);
+		}
+		if ( zeroComments ) {
+			return (
+				<EmptyContent
+					illustration="/calypso/images/comments/illustration_comments_gray.svg"
+					illustrationWidth={ 150 }
+					key="comment-list-empty"
+					line={ emptyMessageLine }
+					title={ emptyMessageTitle }
+				/>
+			);
+		}
+		return null;
+	}
```

Comment:
seems like we're not adding much by extracting this, maybe adding some awkwardness with the `null`

we're basically trapping on two values: do we have comments? should we?

```js
render() {
	const { comments } = this.props;
	…
	return {
		…
		{ zeroComments && hasLoaded && (
			<CommentDetailPlaceholder … />
		) }
		{ zeroComments && ! hasLoaded && (
			<EmptyContent
				…
			/>
		) }
		…
	};
}
```

thoughts? did you find much gain by pulling it out? it seems to hide in the main render area what goes there